jwst/assign_wcs/util.py

```diff
@@ -88,7 +90,81 @@ def _reproject(x, y):
     return _reproject
 
 
-def wcs_from_footprints(dmodels, refmodel=None, transform=None, bounding_box=None, domain=None):
+def compute_scale(wcs: WCS, fiducial: Union[tuple, np.ndarray]) -> float:
+    """Compute scaling transform.
+
+    Parameters
+    ----------
+    wcs :
+        Reference WCS object from which to compute a scaling factor.
+
+    fiducial :
+        Input fiducial of (RA, DEC) used in calculating reference points.
+
+    Returns
+    -------
+    scale :
+        Scaling factor for x and y.
+
+    """
+    if len(fiducial) != 2:
+        raise ValueError(f'Input fiducial must contain only (RA, DEC); Instead recieved: {fiducial}')
+
+    crpix = np.array(wcs.backward_transform(*fiducial))
+    crpix_with_offsets = np.vstack((crpix, crpix + (1, 0), crpix + (0, 1))).T
+    crval_with_offsets = wcs(*crpix_with_offsets)
+
+    coords = SkyCoord(ra=crval_with_offsets[0], dec=crval_with_offsets[1], unit="deg")
+    xscale = np.abs(coords[0].separation(coords[1]).value)
+    yscale = np.abs(coords[0].separation(coords[2]).value)
+
+    return np.sqrt(xscale * yscale)
+
+
+def calc_rotation_matrix(angle: float, v3i_yang: float, vparity: int = 1) -> List[float]:
+    """Calculate the rotation matrix.
+
+    Parameters
+    ----------
+    angle :
```

Comment:
It might be worth either calling this `angle` `roll_ref` instead, or indicating that that is the angle that one needs here.  I.e. the position angle of the V3 axis of the telescope translated to the aperture reference point.

From the JWST keyword dictionary for ROLL_REF:

"Telescope roll angle of V3 North over East at the ref. point (deg)"